tekton/pipeline_run.go

```diff
@@ -123,13 +129,34 @@ func (r *ReleasePipelineRun) WithOwner(release *v1alpha1.Release) *ReleasePipeli
 
 // WithReleaseLabels adds Release name, namespace, and workspace as labels to the release PipelineRun.
 func (r *ReleasePipelineRun) WithReleaseLabels(releaseName, releaseNamespace, releaseWorkspace string) *ReleasePipelineRun {
-	r.ObjectMeta.Labels = map[string]string{
+	releaseLabels := map[string]string{
 		PipelinesTypeLabel:    PipelineTypeRelease,
 		ReleaseNameLabel:      releaseName,
 		ReleaseNamespaceLabel: releaseNamespace,
 		// PipelineRun does not allow labels with : in the value, which KCP workspaces have
 		ReleaseWorkspaceLabel: strings.ReplaceAll(releaseWorkspace, ":", "__"),
 	}
+	// makes sure previously added labels are kept
+	if len(r.ObjectMeta.Labels) != 0 {
+		for k, v := range releaseLabels {
+			r.ObjectMeta.Labels[k] = v
+		}
+	} else {
+		r.ObjectMeta.Labels = releaseLabels
+	}
+
+	return r
+}
+
+// WithApplicationLabel adds the Application name as label to the release PipelineRun.
+func (r *ReleasePipelineRun) WithApplicationLabel(applicationName string) *ReleasePipelineRun {
+	if len(r.ObjectMeta.Labels) == 0 {
+		r.ObjectMeta.Labels = map[string]string{
+			ApplicationNameLabel: applicationName,
+		}
+	} else {
+		r.ObjectMeta.Labels[ApplicationNameLabel] = applicationName
```

Comment:
This line should be enough. Just make sure that an empty release already have an empty map for the labels so the other functions work.